File: src/main/scala/com/github/mauricio/postgresql/util/DaemonThreadsFactory.scala
package com.github.mauricio.postgresql.util

import java.util.concurrent.{Executors, ThreadFactory}


/**
 * User: Maurício Linhares
 * Date: 3/3/12
 * Time: 2:22 PM
 */

object DaemonThreadsFactory extends ThreadFactory {
  def newThread(r: Runnable): Thread = {

    val thread = Executors.defaultThreadFactory().newThread(r)
    thread.setDaemon(true)

    return thread
  }
}

File: db-async-common/src/main/java/com/github/jasync/sql/db/util/DaemonThreadsFactory.kt
package com.github.jasync.sql.db.util

import java.util.concurrent.Executors
import java.util.concurrent.ThreadFactory
import java.util.concurrent.atomic.AtomicInteger

data class DaemonThreadsFactory(val name: String) : ThreadFactory {

    private val threadNumber = AtomicInteger(1)

    override fun newThread(r: Runnable): Thread {
        val thread = Executors.defaultThreadFactory().newThread(r)
        thread.isDaemon = true
        val threadName = name + "-thread-" + threadNumber.getAndIncrement()
        thread.name = threadName
        return thread
    }
}
